engine/include/cubos/engine/cubos.hpp

```diff
@@ -49,44 +50,61 @@ namespace cubos::engine
         /// @return Reference to this object, for chaining.
         template <typename C> Cubos& addComponent();
 
-        /// Adds a new system to the engine, which will be executed every iteration of the main loop.
-        /// If the stage doesn't already exist, it is created in the previously specified default position.
+        /// Sets the current tag for the main dispatcher. If the tag doesn't exist, it will be created.
+        /// Subsequent calls will set this tag's settings.
+        /// @param tag The tag to set.
+        Cubos& tag(const std::string& tag);
```

Comment:
Maybe instead of returning `Cubos` we should return a `TagBuilder` (which keeps a reference to `Cubos`), for which we can then call the property methods (`tagged`, `after`, `before`, ...). This way, we prevent some possible errors at compile time.

The same goes for the systems (e.g, returning a SystemBuilder).

Reply:
How should users "escape" the scope of a builder? Should we have users save the Cubos object so they can access it later for other operations?

```cpp
Cubos c;
c.tag("A").before("B"); // TagBuilder
c.system(a).tagged("A"); // SystemBuilder
c.run(); // Cubos
```

Reply:
Yeah, I think this is fine. On plugins we already have the object being passed as argument, so its not an issue there.

Reply:
Implemented